src/core/src/core_logic/PatchInstaller.py

```diff
@@ -88,6 +90,15 @@ def start_installation(self, simulate=False):
 
         self.composite_logger.log("\nInstalled update count: " + str(installed_update_count) + " (including dependencies)")
 
+        endTime = self.env_layer.datetime.datetime_utcnow()
+        patchService = self.package_manager.__class__.__name__
+        machine_details = TelemetryWriter.machine_info
+
+        InstallUpdatesDetails = {'job': "Install Updates", 'startTime': str(startTime), 'endTime': str(endTime), 'installed_update_count': str(installed_update_count), 'update_run_successful': str(update_run_successful),
```

Comment:
I'd suggest doing the time diff right here (instead of taking that to kusto), and also measuring what percentage of the maintenance window was consumed.

Reply:
done